src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -743,6 +743,13 @@ protected function embed_links( $data, $embed = true ) {
 						continue;
 					}
 
+					if ( empty( $request['per_page'] ) ) {
+						$matched = $this->match_request_to_handler( $request );
```

Comment:
@spacedmonkey, this mutates the `$request` and sets the default `context=view`, which prevents the code below from setting it to an `embed`.

This is the cause of navigation fallback regression - https://core.trac.wordpress.org/ticket/43439#comment:17.